portal-kernel/src/com/liferay/portal/kernel/upgrade/GuestUnsupportedResourceActionsUpgradeProcess.java

```diff
@@ -0,0 +1,149 @@
+				bitmask, companyId));
+	}
+
+	private long _getBitmask() throws Exception {
+		long bitmask = 0xFFFFFFFFFFFFFFFFL;
+
+		StringBundler sb = new StringBundler(
+			2 + _guestUnsupportedResourceActions.length);
+
+		sb.append("select bitwiseValue from ResourceAction where name = ? ");
+		sb.append("and (");
+
+		for (int i = 1; i < _guestUnsupportedResourceActions.length; i++) {
+			sb.append("actionId = ? or ");
+		}
+
+		sb.append("actionId = ?)");
+
+		try (PreparedStatement preparedStatement = connection.prepareStatement(
+				sb.toString())) {
+
+			preparedStatement.setString(1, _resourceName);
+
+			for (int i = 0; i < _guestUnsupportedResourceActions.length; i++) {
+				preparedStatement.setString(
+					i + 2, _guestUnsupportedResourceActions[i]);
+			}
+
+			try (ResultSet resultSet = preparedStatement.executeQuery()) {
+				while (resultSet.next()) {
+					bitmask &= ~resultSet.getLong("bitwiseValue");
+				}
+			}
+		}
+
+		return bitmask;
+	}
+
+	private long _getGuestRoleId(long companyId) throws Exception {
+		try (PreparedStatement preparedStatement = connection.prepareStatement(
+				"select roleId from Role_ where companyId = ? and name = ?")) {
+
+			preparedStatement.setLong(1, companyId);
+			preparedStatement.setString(2, RoleConstants.GUEST);
+
+			try (ResultSet resultSet = preparedStatement.executeQuery()) {
+				if (!resultSet.next()) {
+					throw new IllegalStateException(
+						"Unable to find Guest role for company " + companyId);
+				}
+
+				return resultSet.getLong("roleId");
+			}
+		}
+	}
+
+	private void _removeGuestUnsupportedResourceActionsPermissions(
+			long bitmask, long companyId)
+		throws Exception {
+
+		long guestRoleId = _getGuestRoleId(companyId);
+
+		try (PreparedStatement preparedStatement1 = connection.prepareStatement(
+				"select resourcePermissionId, actionIds from " +
+					"ResourcePermission where companyId = ? and name = ? and " +
+						"roleId = ?")) {
+
+			preparedStatement1.setLong(1, companyId);
+			preparedStatement1.setString(2, _resourceName);
+			preparedStatement1.setLong(3, guestRoleId);
+
+			try (ResultSet resultSet = preparedStatement1.executeQuery();
+				PreparedStatement preparedStatement2 =
+					AutoBatchPreparedStatementUtil.concurrentAutoBatch(
+						connection,
+						"update ResourcePermission set actionIds = ? where " +
+							"resourcePermissionId = ?")) {
+
+				while (resultSet.next()) {
+					preparedStatement2.setLong(
+						1, resultSet.getLong("actionIds") & bitmask);
+					preparedStatement2.setLong(
+						2, resultSet.getLong("resourcePermissionId"));
+
+					preparedStatement2.addBatch();
+				}
+
+				preparedStatement2.executeBatch();
+			}
+		}
+
+		if (!ArrayUtil.contains(
+				_guestUnsupportedResourceActions, ActionKeys.VIEW)) {
+
+			return;
+		}
+
+		try (PreparedStatement preparedStatement = connection.prepareStatement(
+				"update ResourcePermission set viewActionId = 0) where " +
```

Comment:
Should the `)` be in `update ResourcePermission set viewActionId = 0) ...`. I don't see any opening one

Reply:
Ah! You're absolutely right. Sorry about that. I didn't test this part I guess since we weren't dealing with the VIEW permission in this case.

Reply:
I've force-pushed a fix for this.

Reply:
Thank you!